File: Energi-Potensial/EnergiPotensial.java
import java.util.Scanner;

public class EnergiPotensial
{
    Scanner scanner = new Scanner (System.in);

    static final float gravitasi = 10;

    float hasil_energi_potensial ()
    {
        float massa = 0, ketinggian = 0;

        System.out.print ("Masukkan massa (satuan kg): ");
        massa = scanner.nextFloat ();

        System.out.print ("Masukkan ketinggian (satuan m): ");
        ketinggian = scanner.nextFloat ();

        return massa * gravitasi * ketinggian;
    }

    float hasil_massa ()
    {
        float energi_potensial = 0, ketinggian = 0;

        System.out.print ("Masukkan energi potensial (satuan Joule): ");
        energi_potensial = scanner.nextFloat ();

        System.out.print ("Masukkan ketinggian (satuan m): ");
        ketinggian = scanner.nextFloat ();

        return energi_potensial / (gravitasi * ketinggian);
    }

    float hasil_ketinggian ()
    {
        float energi_potensial = 0, massa = 0;

        System.out.print ("Masukkan energi potensial (satuan Joule): ");
        energi_potensial = scanner.nextFloat ();

        System.out.print ("Masukkan massa (satuan kg): ");
        massa = scanner.nextFloat ();

        return energi_potensial / (massa * gravitasi);
    }

    public static void main (String [] args)
    {
        EnergiPotensial ep = new EnergiPotensial ();

        byte pilihan = 0;
        float massa = 0, ketinggian = 0, energi_potensial = 0;

        System.out.println ("========== Penghitung Energi Potensial ==========");

        do
        {
            System.out.println ("1. Mencari energi potensial");
            System.out.println ("2. Mencari massa");
            System.out.println ("3. Mencari ketinggian");
            System.out.println ("4. Keluar");
            System.out.print ("Pilih salah satu: ");
            pilihan = ep.scanner.nextByte ();

            switch (pilihan)
            {
                case 1:
                    energi_potensial = ep.hasil_energi_potensial ();
                    System.out.println ("Energi potensial: " + energi_potensial + " Joule");
                    break;
                case 2:
                    massa = ep.hasil_massa ();
                    System.out.println ("Massa: " + massa + " kg");
                    break;
                case 3:
                    ketinggian = ep.hasil_ketinggian ();
                    System.out.println ("Ketinggian: " + ketinggian + " m");
                    break;
                case 4:
                    System.out.println ("Terima kasih telah menggunakan program ini");
                    break;
                default:
                    System.out.println ("Input error");
                    break;
            }
        } while (pilihan < 1 || pilihan > 4);
    }
}

File: Energi Potensial/Energi-Potensial.c
#include <stdio.h>

const float gravitasi = 10;

float hasil_massa ();
float hasil_ketinggian ();
float hasil_energi_potensial ();

int main()
{
    unsigned short pilihan = 0;
    float massa = 0, ketinggian = 0, energi_potensial = 0;

    do
    {
        puts ("========== Penghitung Energi Potensial ==========");
        puts ("1. Mencari energi potensial");
        puts ("2. Mencari massa");
        puts ("3. Mencari ketinggian");
        puts ("4. Keluar");
        printf ("Pilih salah satu: ");
        scanf ("%hu", &pilihan);

        switch (pilihan)
        {
            case 1:
                energi_potensial = hasil_energi_potensial ();
                printf ("Energi potensial: %f Joule\n", energi_potensial);
                break;
            case 2:
                massa = hasil_massa ();
                printf ("Massa: %f kg\n", massa);
                break;
            case 3:
                ketinggian = hasil_ketinggian ();
                printf ("Ketinggian: %f m\n", ketinggian);
                break;
            case 4:
                puts ("Terima kasih telah menggunakan program ini!");
                break;
            default:
                puts ("Input error");
        }
    } while (pilihan != 4);

    return 0;
}

float hasil_energi_potensial ()
{
    float massa = 0, ketinggian = 0;

    printf ("Masukkan massa (satuan kg): ");
    scanf ("%f", &massa);

    printf ("Masukkan ketinggian (satuan m): ");
    scanf ("%f", &ketinggian);

    return massa * gravitasi * ketinggian;
}

float hasil_massa ()
{
    float energi_potensial = 0, ketinggian = 0;

    printf ("Masukkan energi potensial (satuan Joule): ");
    scanf ("%f", &energi_potensial);

    printf ("Masukkan ketinggian (satuan m): ");
    scanf ("%f", &ketinggian);

    return energi_potensial / (gravitasi * ketinggian);
}

float hasil_ketinggian ()
{
    float energi_potensial = 0, massa = 0;

    printf ("Masukkan energi potensial (satuan Joule): ");
    scanf ("%f", &energi_potensial);

    printf ("Masukkan massa (satuan kg): ");
    scanf ("%f", &massa);

    return energi_potensial / (massa * gravitasi);
}